java-manta-client/src/main/java/com/joyent/manta/client/multipart/EncryptedMultipartUpload.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
+ */
+package com.joyent.manta.client.multipart;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Multipart upload state object used with client-side encryption.
+ *
+ * @author <a href="https://github.com/dekobon">Elijah Zupancic</a>
+ * @since 3.0.0
```

Comment:
Some sort of explanation for why the `EncryptionState` is here, and why you should/can not touch it.